api/app/controllers/spree/api/stock_items_controller.rb

```diff
@@ -54,7 +54,11 @@ def destroy
       private
 
       def load_stock_location
-        render 'spree/api/shared/stock_location_required', status: 422 and return unless params[:stock_location_id]
+        if !params[:stock_location_id]
```

Comment:
Here I think we could use `params.fetch(:stock_location_id)` to get an `ActionController::ParameterMissing` instead of writing this logic ourselves